apps/src/dance/ai/DanceAiEmojiIcon.tsx

```diff
@@ -0,0 +1,38 @@
+import React, {useCallback} from 'react';
+
+import EmojiIcon from './EmojiIcon';
+
+type AiModalItem = {
+  id: string;
+  emoji: string;
+};
+
+interface DanceAiEmojiIconProps {
+  item: AiModalItem;
+  handleItemClick?: (itemId: string, isItemAvailable: boolean) => void;
+  className?: string;
+  isItemAvailable?: boolean;
+}
+
+const DanceAiEmojiIcon: React.FunctionComponent<DanceAiEmojiIconProps> = ({
+  item,
+  handleItemClick = () => undefined,
+  className,
+  isItemAvailable = true,
+}) => {
+  const danceAiClickHandler = useCallback(
+    () => handleItemClick(item.id, isItemAvailable),
+    [handleItemClick, item.id, isItemAvailable]
+  );
+
+  return (
+    <EmojiIcon
+      item={item}
+      onClick={danceAiClickHandler}
+      className={className}
+      isHighlighted={!isItemAvailable}
+    />
+  );
+};
+
+export default React.memo(DanceAiEmojiIcon);
```

Comment:
this component is a wrapper around the more generic `EmojiIcon`.

Each `DanceAiEmojiIcon` gets a custom click handler which is passed the item's id and its availability.

This way we can memoize the component and render properly. Otherwise we'd be creating a new click handler every render, and need to re-draw.

mid-long term, this could be replaced by a new click handler which doesn't pass along the `isItemAvailable` flag (it can be looked up from the `selectedEmojis` and then gets the ID from a data-tag. But we can circle back to that.